api/src/modules/scenarios/wdpa-arena-calculation.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@nestjs/common';
+import { pick } from 'lodash';
+import { CreateScenarioDTO } from './dto/create.scenario.dto';
+import { UpdateScenarioDTO } from './dto/update.scenario.dto';
+import { Scenario } from './scenario.api.entity';
+
+type InputChange = CreateScenarioDTO | UpdateScenarioDTO;
+
+@Injectable()
+export class WdpaArenaCalculationService {
+  /**
+   * if one of those is located within Input Change
+   */
+  private readonly watchedChangeProperties: (keyof InputChange)[] = [
+    'customProtectedAreaIds',
+    'wdpaIucnCategories',
+    'wdpaThreshold',
+  ];
+
+  /**
+   * and new entity state consists (i.e. are present) of all of the below
+   */
+  private readonly requiredToTriggerChange: (keyof Scenario)[] = [
+    'wdpaThreshold',
+  ];
+
+  /**
+   * Every post update that affects this 3 elements (when threshold is not null)
+   */
+  shouldTrigger(scenario: Scenario, changeSet: InputChange): boolean {
+    if (!this.intendsToChangeWatchedProperty(changeSet)) {
+      return false;
+    }
+
+    return this.areRequiredFieldsAvailable(scenario);
+  }
+
+  private intendsToChangeWatchedProperty(changeSet: InputChange): boolean {
+    return Object.entries(pick(changeSet, this.watchedChangeProperties)).some(
+      ([, value]) => value,
+    );
+  }
+
+  private areRequiredFieldsAvailable(scenario: Scenario): boolean {
+    return Object.entries(pick(scenario, this.requiredToTriggerChange)).every(
+      ([, value]) => value,
+    );
+  }
```

Comment:
these seem like good candidates to move to utility functions (when we need these kinds of checks for a second context) - just a note, nothing to do now